lib/routes/routeBackbeat.js

```diff
@@ -851,7 +851,11 @@ function routeBackbeat(clientIP, request, response, log) {
                 return backbeatRoutes[request.method][request.resourceType]
                     [request.query.operation](request, response, log, next);
             }
-            if (!bucketInfo.isNFS() && !bucketInfo.isVersioningEnabled()) {
+            const allowNonVersionedBucket =
```

Comment:
I agree. We do enforce versioning when setting a replication config, but there is no check in case versioning is suspended later on. Ultimately, I think the best option is to remove this check and add a check for suspending versioning on buckets with a replication config enabled. That way we can ensure that any bucket with replication will meet this condition, and lifecycle operations can work regardless.